docker-compose.yml

```diff
@@ -11,7 +11,7 @@ services:
       # and must list all the names under which the puppetserver can be
       # reached. 'puppet' must be one of them, otherwise puppetdb won't be
       # able to get a cert. Add other names as a comma-separated list
-      - DNS_ALT_NAMES
+      - DNS_ALT_NAMES=puppet,${DNS_ALT_NAMES:-}
```

Comment:
So it looks like setting a default like this `${DNS_ALT_NAMES:-}` suppresses the warning.

More info in https://docs.docker.com/compose/compose-file/#variable-substitution

Reply:
I guess only `.env` can be consumed inside the compose files. `env_file` is supposed to be for injecting env vars into the container apparently:

https://docs.docker.com/compose/environment-variables/#configure-compose-using-environment-variables
https://github.com/docker/compose/issues/4670

Reply:
Gah, duh! Why didn't we try this first 😄